numericValidation-fxdDepProd

Jov03 opened the pull request:
## Description
Added decimal-number attribute to the numeric validation directive( late-validate directive) to validate fields in fixed New Deposit product view

## Related issues and discussion
Issue No. #2794 Previous PR #2809 

## Screenshot
![screenshot from 2018-12-28 14-06-34](https://user-images.githubusercontent.com/15383669/50508977-2308ad00-0aaa-11e9-82bb-f4165c383952.png)

- [x] Validate the JS and HTML files with `grunt validate` to detect errors and potential problems in JavaScript code.

- [x] Run the tests by opening `test/SpecRunner.html` in the browser to make sure you didn't break anything.

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `community-app/Contributing.md`.

abhaychawla reviewed (commented):
@Jov03 Looks good. Please update your commit message as per the guidelines to `Fix #issueno - Issue Description` and also mention the related issue id in your PR description so I can approve.

Jov03 commented:
@abhaychawla I updated the PR.

abhaychawla reviewed (approved):
Thanks!

abhaychawla commented:
@ShruthiRajaram please merge this PR.

ShruthiRajaram commented on app/views/products/createfixeddepositproduct.html:
```diff
@@ -94,107 +100,102 @@
                         <div class="col-sm-2">
                             <input type="text" id="depositAmount" name="depositAmount" class="form-control"
-                                   number-format ng-model="formData.depositAmount" required late-validate/>
+                                number-format ng-model="formData.depositAmount" required 
```
deposit amount should include decimal values also . Issue reference -2985

Jov03 replied:
@ShruthiRajaram  ok ill make the necessary changes.

ShruthiRajaram commented on app/views/products/createfixeddepositproduct.html:
```diff
@@ -94,107 +100,102 @@
                             <input type="text" id="minDepositAmount" name="minDepositAmount" class="form-control"
-                                   number-format ng-model="formData.minDepositAmount" late-validate/>
-                            <form-validate valattributeform="Terms" valattribute="minDepositAmount"/>
+                                number-format ng-model="formData.minDepositAmount" late-validate number=true />
```
same as deposit amount

ShruthiRajaram commented on app/views/products/createfixeddepositproduct.html:
```diff
@@ -94,107 +100,102 @@
-                                   number-format ng-model="formData.maxDepositAmount" late-validate/>
-                            <form-validate valattributeform="Terms" valattribute="maxDepositAmount"/>
+                                number-format ng-model="formData.maxDepositAmount" late-validate number=true />
+                            <form-validate valattributeform="Terms" valattribute="maxDepositAmount" />
```
same as above

Jov03 committed:
#2794-numeric Validation in Create Fixed Deposit Product View

Added Numeric Validation to the textfields in Create Fixed Deposit Product View .Added decimal number attribute to late-validate directive to validate decimal numbers.

Jov03 commented:
@abhaychawla @ShruthiRajaram  I looked into issue 2985 as requested the decimal directive used there was updating the user input automatically not validating the textfield so i added decimal number attribute to existing late-validate directive to validate the textfields. I have attached a screenshot in PR description please have a look.

On a side note Travis Build is failing on inspecting the logs generated by travis it is due to connect.js file to which i have not made any changes but let me know if it is due to my changes to the code.

abhaychawla commented:
This is not being caused by your changes. It seems the Travis build is failing as the version of node js being used for the build doesn't support the ES6 syntax. If possible, please create an issue for the same and send a PR to update the version?

@ShruthiRajaram, as @Jov03 said above a validation message is not being shown where decimals directive is being used rather the field gets updated automatically, is it deliberate or a different approach should be followed as in the updated PR?

ShruthiRajaram commented:
@Jov03 raise the PR again and try to get a successful Travis build. once Travis build is success will merge the PR.

Jov03 commented:
@ShruthiRajaram i raised the PR again and build failed due to same reason as explained in issue #3077. Please have a look.

Jov03 commented:
Will close this PR and re-open it to see if it makes any difference.

Jov03 commented:
@ShruthiRajaram @abhaychawla after recent PR#3087 merge my PR passed  Travis build please have a look.
